SCons/Scanner/Java.py

```diff
@@ -21,56 +21,58 @@
 # OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 # WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-import os 
+import os
 
 import SCons.Node
 import SCons.Node.FS
 import SCons.Scanner
-import SCons.Util
+from SCons.Util import flatten, is_String
 
 
-def _subst_libs(env, libs):
-    """
-    Substitute environment variables and split into list.
+def _subst_paths(env, paths) -> list:
+    """Return a list of substituted path elements.
+
+    If *paths* is a string, it is split on the search-path separator
+    (this makes the interpretation system-specitic - this is warned about
+    in the manpage).  This helps support behavior like pulling in the
+    external ``CLASSPATH`` and setting it directly into ``JAVACLASSPATH``.
+    Otherwise, substitution is done on string-valued list elements
+    but not splitting.
     """
-    if SCons.Util.is_String(libs):
-        libs = env.subst(libs)
-        if SCons.Util.is_String(libs):
-            libs = libs.split()
-    elif SCons.Util.is_Sequence(libs):
-        _libs = []
-        for lib in libs:
-            _libs += _subst_libs(env, lib)
-        libs = _libs
+    if is_String(paths):
+        paths = env.subst(paths)
+        if SCons.Util.is_String(paths):
+            paths = paths.split(os.pathsep)
     else:
-        # libs is an object (Node, for example)
-        libs = [libs]
-    return libs
+        paths = flatten(paths)
+        paths = [env.subst(path) if is_String(path) else path for path in paths]
```

Comment:
This won't handle a sequence with strings each of which had pathsep separated jars, which would have worked before?

Reply:
hmm. maybe stick a note there in the code as a TODO reevaluate if bugs are filed (a note to future self)?